### PR TITLE
INT-2562 resilience for red hat trigger

### PR DESCRIPTION
**Pull request by jflinchbaugh**
1. allow red hat builds to be skipped in case the service is down
2. retry on failure of service while watching for build to finish.

JIRA: https://issues.sonatype.org/browse/INT-2562

### CODE REVIEW

File: ci/TriggerRedHatBuild.groovy [hunk 1]
```diff
@@ -14,7 +14,8 @@ import groovyx.net.http.HttpBuilder
 import groovyx.net.http.HttpException
 
 if (args.size() < 3) {
-  fail('Usage: groovy TriggerRedhatBuild.groovy <version> <projectId> <apiKey>')
+  System.err.println('Usage: groovy TriggerRedhatBuild.groovy <version> <projectId> <apiKey>')
```
Comment (kellyrob99) [inline ci/TriggerRedHatBuild.groovy:17]: why is this change required please? Couldn't immediately find a reference to this 'fail' method, but this might fit the bill better: https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#error-error-signal

File: ci/TriggerRedHatBuild.groovy [hunk 2]
```diff
@@ -145,12 +146,17 @@ class BuildClient {
...
+        final completedBuild = getTags().find {
+          it.name == nextTag && it.scan_status == 'passed'
+        }
+
+        if (completedBuild) {
+          return completedBuild
+        }
+      } catch (HttpException ex) {
+        ex.printStackTrace()
+        System.err.println "Failed waiting for complete build, but still retrying: ${ex.statusCode} [${ex.body}]"
```
Comment (kellyrob99) [inline ci/TriggerRedHatBuild.groovy:159]: why not 'echo' for printing out messages?

File: ci/TriggerRedHatBuild.groovy (hunk 1 above)
Comment (kellyrob99) [reply ci/TriggerRedHatBuild.groovy:17]: nm - I see this is no longer a Jenkins script :)

Comment (jflinchbaugh) [reply ci/TriggerRedHatBuild.groovy:17]: yeah, this is a stand-alone to which jenkins delegates.

Comment (jflinchbaugh) [reply ci/TriggerRedHatBuild.groovy:17]: also, I changed it because I found out that the fail() method wasn't available as I had tried to use it earlier. (bug!)
